src/randomeffectsterm.jl

```diff
@@ -3,43 +3,16 @@ abstract type AbstractReTerm <: AbstractTerm end
 struct RandomEffectsTerm <: AbstractReTerm
     lhs::StatsModels.TermOrTerms
     rhs::StatsModels.TermOrTerms
-    function RandomEffectsTerm(lhs, rhs)
-        if isempty(intersect(StatsModels.termvars(lhs), StatsModels.termvars(rhs)))
-            if !isa(
-                rhs,
-                Union{
-                    CategoricalTerm,
-                    InteractionTerm{<:NTuple{N,CategoricalTerm} where {N}},
-                },
-            )
-                throw(ArgumentError("blocking variables (those behind |) must be Categorical ($(rhs) is not)"))
-            end
-            new(lhs, rhs)
-        else
-            throw(ArgumentError("Same variable appears on both sides of |"))
-        end
-    end
 end
 
-function StatsModels.apply_schema(
-    t::FunctionTerm{typeof(/)},
-    sch::StatsModels.FullRank,
-    Mod::Type{<:MixedModel},
-)
-    if length(t.args_parsed) ≠ 2
-        throw(ArgumentError("malformed nesting term: $t (Exactly two arguments required"))
-    end
-
-    first, second = apply_schema.(t.args_parsed, Ref(sch), Mod)
-    
-    if !(typeof(first) <: CategoricalTerm)
-        throw(ArgumentError("nesting terms requires categorical grouping term, got $first.  Manually specify $first as `CategoricalTerm` in hints/contrasts"))
-    end
+# TODO: consider overwriting | with our own function that can be
+# imported with (a la FilePathsBase.:/)
+# using MixedModels: |
+# to avoid conflicts with definitions in other packages...
+Base.:|(a::StatsModels.TermOrTerms, b::StatsModels.TermOrTerms) = RandomEffectsTerm(a, b)
 
-    return first + fulldummy(first) & second
-end
-
-RandomEffectsTerm(lhs, rhs::NTuple{2,AbstractTerm}) =
+# expand (lhs | a + b) to (lhs | a) + (lhs | b)
```

Comment:
what happens with the nesting `/` syntax when constructed programmatically?

Reply:
(I mostly mean in the grouping/block vars, i.e. RHS of `|`, but would also be curious about LHS)

Reply:
Nothing right now (it won't work AFAIK) since we don't define methods for `Base.:/(a::AbstractTerm, b::AbstractTerm)`

Reply:
but I guess that's now the odd-one-out so maybe worth it?

Reply:
If you can do it in a few lines, please. :smile: We have a certain user who's a big fan. :wink:

Reply:
actually this is a bit trickier than I thought at first...the run-time stuff has to be able to work before we have an applied schema, so we'd have to either be able to lazily convert to full-dummy, or lazily expand the `/`.

OR, actually, maybe it would be okay since the full dummy thing only applies to fixed effects anyway and this is exactly what the `FullRank` schema is intended to handle (since if you have `1 + a + a & b`, there's no `b`, so `a` should get promoted).

The other possibly slightly tricky thing would be deciding whether to apply the "distributive rule" for `|` at run-time or leave that until schema time; but I think that's handled by a constructor right now so should also be okay.

Reply:
I'll leave it up to you. If you think it's not worthwhile, I would add one line to the docs and move on.